### PR TITLE
test: Enable scripts to be referenced in configMapGenerator and Enable Sweeper

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

This PR adds a sweeper cron which runs the cleanup script every day to delete any instances that are older than 12 hours, any stacks older than 12 hours, and any launch templates older than 12 hours.

**How was this change tested?**

* Manual run of the scripts

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note
None
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Clean up the CI account from failed CF stacks or failed test runs
+# The script assumes the aws CLI command is authenticated AWS_REGION is set
+
+# EXPIRATION_TIME is the exact time in UTC that resources expire and will be swept by the script
+EXPIRATION_TIME=$(date -d 'now-12 hours' +%Y-%m-%dT%H:%M --utc)
+echo "Expiration time for all resources is: $EXPIRATION_TIME"
+
+# Grab all instances that are older than the EXPIRATION_TIME and then filter out the KITInfrastructure instances
+old_instances=($(aws ec2 describe-instances --filters Name=instance-state-name,Values=running --query "Reservations[].Instances[?LaunchTime <= '$EXPIRATION_TIME'][]" | grep -v "kubernetes.io/cluster/KITInfrastructure" | jq -r '.[].InstanceId' | sort))
```
Comment (njtran) [inline test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh:11]: nice bash and awscli logic :D

File: test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+n_old_instances="${#old_instances[@]}"
+echo "Removing ${n_old_instances} old instances"
+
+if (( n_old_instances > 0 )); then
+  aws ec2 terminate-instances --instance-ids ${old_instances}
```
Comment (njtran) [inline test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh:16]: Any thought on doing the following so one failure doesn't stop the rest of the script? 
```suggestion
  aws ec2 terminate-instances --instance-ids ${old_instances} || true
```

File: test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# Delete old stacks from the account
+stacks=($(aws cloudformation list-stacks --stack-status-filter CREATE_COMPLETE ROLLBACK_FAILED ROLLBACK_COMPLETE DELETE_FAILED UPDATE_COMPLETE UPDATE_FAILED UPDATE_ROLLBACK_FAILED UPDATE_ROLLBACK_COMPLETE IMPORT_COMPLETE IMPORT_ROLLBACK_FAILED IMPORT_ROLLBACK_COMPLETE --query "StackSummaries[?CreationTime <= '$EXPIRATION_TIME']" | jq -r '.[].StackName' | grep 'karpenter-tests-*-*'))
+n_stacks="${#stacks[@]}"
+echo "Removing ${n_stacks} cloudformation stacks"
+
+# Iterate through every line in the stack describe response
+for stack in "${stacks[@]}"; do
+  echo "Deleting Stack ${stack}"
+  aws cloudformation delete-stack --no-cli-pager --stack-name "${stack}"
```
Comment (njtran) [inline test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh:27]: Any thought on doing the following so one failure doesn't stop the rest of the script? Same here
```suggestion
  aws cloudformation delete-stack --no-cli-pager --stack-name "${stack}" || true
```

File: test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+# Clean up old launch templates
+launch_templates=($(aws ec2 describe-launch-templates --query "LaunchTemplates[?CreateTime <= '$EXPIRATION_TIME']" |  jq -r '.[].LaunchTemplateName' | grep "karpenter.k8s.aws"))
+n_lts="${#launch_templates[@]}"
+echo "Removing ${n_lts} launch templates"
+
+# Iterate through every line in the launch template describe response
+for lt in "${launch_templates[@]}"; do
+  echo "Deleting LT ${lt}"
+  aws ec2 delete-launch-template --no-cli-pager --launch-template-name "${lt}"
```
Comment (njtran) [inline test/infrastructure/clusters/test-infra/karpenter-tests/scripts/cleanup.sh:38]: Same here

```suggestion
  aws ec2 delete-launch-template --no-cli-pager --launch-template-name "${lt}" || true
```

File: test/infrastructure/clusters/test-infra/karpenter-tests/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  - sweeper-cron.yaml
+  - rbac.yaml
+  - test-suites.yaml
+configMapGenerator:
+  - name: scripts
+    namespace: karpenter-tests
+    files:
+      - scripts/pipeline-cleanup.sh
+      - scripts/pipeline-trigger.sh
+      - scripts/sweep.sh
```
Comment (njtran) [inline test/infrastructure/clusters/test-infra/karpenter-tests/kustomization.yaml:15]: should this match the file name?
```suggestion
      - scripts/cleanup.sh
```

Comment (jonathan-innis) [reply test/infrastructure/clusters/test-infra/karpenter-tests/kustomization.yaml]: Good catch

**Review by njtran — APPROVED** [review]

**Pull request merged**